### PR TITLE
pkp/pkp-lib#9658 send and accept invitation views and php integrations

### PR DESCRIPTION
**Pull request by ipula**

### CODE REVIEW

File: locale/en/invitation.po [hunk 1]
```diff
@@ -101,4 +101,310 @@ msgid "invitation.userRoleAssignment.userGroup.startDate.mustBeAfterToday"
...
+msgstr "Accept And Continue to OJS"
+
+msgid "acceptInvitation.skipVerifyOrcid"
+msgstr "Skip ORCID verification"
+
+msgid "acceptInvitation.verifyOrcid"
+msgstr "Verify ORCID iD"
+
+msgid "acceptInvitation.usernameField.description"
+msgstr "It should be 10 characters long and could be a combination of uppercase letters, lowercase letters or numbers"
```
[the remark this answers is not in the document]
Comment (ipula) [reply locale/en/invitation.po:284]: its happens [here](https://github.com/ipula/ui-library/blob/d5e96c7853892d9dc5cfaaf51e395974a070ca89/src/pages/acceptInvitation/AcceptInvitationUserAccountDetails.vue#L14)

File: locale/en/invitation.po [hunk 2]
```diff
@@ -101,4 +101,310 @@ msgid "invitation.userRoleAssignment.userGroup.startDate.mustBeAfterToday"
...
+
+msgid "acceptInvitation.passwordField.description"
+msgstr "It should be 12 characters long and should be a combination of uppercase letters, lowercase letters, numbers and symbols"
```
[the remark this answers is not in the document]
Comment (ipula) [reply locale/en/invitation.po:287]: same as above.Now its only required 6 characters. I will ask @Devika008 for a new message

**Comment by withanage** [comment]
@ewhanson 
> * The phrase "OJS" and "journal" are used throughout the locale keys. I've indicated these ones with comments, but I think for this PR it would probably be best to get this merged, and address that separately afterwards. So you can keep the comments there, but I wouldn't worry about them right now.
> 

The OJS specific locale were moved to OJS locale, so that other PKP applications may customize them easily.
https://github.com/pkp/ojs/pull/4497

File: locale/en/invitation.po [hunk 3]
```diff
@@ -101,4 +101,310 @@ msgid "invitation.userRoleAssignment.userGroup.startDate.mustBeAfterToday"
...
+msgstr "The :attribute field is prohibited"
+
+msgid "invitation.cancelInvite.actionName"
+msgstr "Cancel Invite"
+
+msgid "invitation.cancelInvite.title"
+msgstr "Cancel Invitation"
+
+msgid "invitation.cancelInvite.message"
+msgstr "Cancel the invitation sent to {$givenName} {$familyName} will deactivate acceptance link sent via email. Here are the invitation details: "
```
[the remark this answers is not in the document]
Comment (ipula) [reply locale/en/invitation.po:365]: https://github.com/orgs/craft-oa/projects/2/views/1?filterQuery=cancel&pane=issue&itemId=75533135&issue=craft-oa%7Cgdpr-invitation%7C15

**Comment by ipula** [comment]
@ewhanson I made some changes to white-spaces and hide next button in ui-library in this [pr](https://github.com/pkp/ui-library/pull/439)

**Comment by ewhanson** [comment]
> The OJS specific locale were moved to OJS locale, so that other PKP applications may customize them easily. [pkp/ojs#4497](https://github.com/pkp/ojs/pull/4497)

Thanks @withanage. Have PRs with equivalents been created for OPS and OMP been created? We'll need those too. Thanks!

File: classes/invitation/stepTypes/SendInvitationStep.php [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+use stdClass;
+
+class SendInvitationStep extends InvitationStepTypes
+{
+    /**
+     * get send invitation steps
+     *
+     * @throws Exception
+     */
+    public function getSteps(?Invitation $invitation, Context $context,?User $user): array
```
Comment (ewhanson) [inline classes/invitation/stepTypes/SendInvitationStep.php:36]: Space needed between `$context` and `?User`.

File: locale/en/invitation.po [hunk 5]
```diff
@@ -101,4 +101,274 @@ msgid "invitation.userRoleAssignment.userGroup.startDate.mustBeAfterToday"
...
+msgstr "Select a new role"
+
+msgid "invitation.role.dateStart"
+msgstr "Start Date"
+
+msgid "invitation.role.dateEnd"
+msgstr "End Date"
+
+msgid "invitation.role.masthead"
+msgstr "Journal Masthead"
```
Comment (ewhanson) [inline locale/en/invitation.po:164]: Anything journal-related should probably move to the applications as well. If you've already done this for the OJS language, could you do this for the word "journal" as well, @withanage?

File: locale/en/invitation.po [hunk 6]
```diff
@@ -101,4 +101,274 @@ msgid "invitation.userRoleAssignment.userGroup.startDate.mustBeAfterToday"
...
+msgstr "Role"
+
+msgid "userInvitation.roleTable.startDate"
+msgstr "Start Date"
+
+msgid "userInvitation.roleTable.endDate"
+msgstr "End Date"
+
+msgid "userInvitation.roleTable.journalMasthead"
+msgstr "Journal Masthead"
```
Comment (ewhanson) [inline locale/en/invitation.po:179]: Usage of journal. Should be moved to app. cc: @withanage.

File: locale/en/invitation.po [hunk 7]
```diff
@@ -101,4 +101,274 @@ msgid "invitation.userRoleAssignment.userGroup.startDate.mustBeAfterToday"
...
+
+msgid "userInvitation.sendMail.stepName"
+msgstr "Review & invite for roles"
+
+msgid "userInvitation.sendMail.nextButtonLabel"
+msgstr "Invite user to the role"
+
+msgid "userInvitation.sendMail.stepDescription"
+msgstr "Send the user an email to let them know about the invitation, next steps, journal GDPR polices and ORCID verification"
```
Comment (ewhanson) [inline locale/en/invitation.po:188]: Usage of journal. Should be moved to app. cc: @withanage.

File: locale/en/invitation.po [hunk 8]
```diff
@@ -101,4 +101,310 @@ msgid "invitation.userRoleAssignment.userGroup.startDate.mustBeAfterToday"
...
+msgstr "Are you sure want to cancel this invitation ?"
+
+msgid "userInvitation.cancel.keepWorking"
+msgstr "Keep Working"
+
+msgid "userInvitation.status.invited"
+msgstr "Invited {$date}"
+
+msgid "userInvitation.search.userNotFound"
+msgstr "The user does not have a role in this journal"
```
[the remark this answers is not in the document]
Comment (ewhanson) [reply locale/en/invitation.po:353]: Usage of journal. Should be moved to app. cc: @withanage.

File: locale/en/invitation.po [hunk 9]
```diff
@@ -101,4 +101,310 @@ msgid "invitation.userRoleAssignment.userGroup.startDate.mustBeAfterToday"
...
+
+msgid "userInvitation.search.userFound"
```
[the remark this answers is not in the document]
Comment (ewhanson) [reply locale/en/invitation.po:355]: Usage of journal. Should be moved to app. cc: @withanage.

File: templates/invitation/acceptInvitation.tpl [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+ * Copyright (c) 2003-2024 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @brief show accept invitation page to the users.
+ *
+ * @hook Template::Settings::access []
+ *}
+{extends file="layouts/backend.tpl"}
+{block name="page"}
+    <accept-invitation-page
```
[the remark this answers is not in the document]
Comment (ewhanson) [reply templates/invitation/acceptInvitation.tpl:15]: These have been converted to spaces. The `.tpl` files should use tabs instead.

File: templates/invitation/userInvitation.tpl [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+ * Copyright (c) 2003-2024 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @brief show create user invitation page to the users.
+ *
+ * @hook Template::Settings::access []
+ *}
+{extends file="layouts/backend.tpl"}
+{block name="page"}
+    <user-invitation-page
```
[the remark this answers is not in the document]
Comment (ewhanson) [reply templates/invitation/userInvitation.tpl:15]: These have been converted to spaces. The `.tpl` files should use tabs instead.

File: templates/management/access.tpl [hunk 12]
```diff
@@ -18,6 +18,7 @@
 
 	<tabs :track-history="true">
 		<tab id="users" label="{translate key="manager.users"}">
+            <user-invitation-manager></user-invitation-manager>
```
[the remark this answers is not in the document]
Comment (ewhanson) [reply templates/management/access.tpl:21]: This line should use tabs like the rest of the file.

**Review by ewhanson — CHANGES_REQUESTED** [review]
Thanks @ipula for the quick changes! Just a few other smaller style things along with a question about journal specific language for @withanage. Once that's taken care of it should be ready for merge!

**Comment by withanage** [comment]
@ewhanson  all the  strings with "Jourals"  were also moved to journal locale

**Review by ewhanson — APPROVED** [review]
Thanks so much, @ipula! Looks good now!

**Pull request merged**